networkx/tests/test_lazy_imports.py

```diff
@@ -0,0 +1,57 @@
+import sys
+import os
+import importlib.util
+import pytest
+
+from networkx import lazy_import, lazy_package_import
+from networkx.lazy_imports import DelayedImportErrorModule
+
+
+def test_lazy_import():
+    np = lazy_import("numpy")
+    anything_not_real = lazy_import("anything_not_real")
+
+    # Now test that accessing attributes does what it should
+    assert np.sin(np.pi) == pytest.approx(0, 1e-6)
+    # poor-mans pytest.raises for testing errors on attribute access
+    try:
+        anything_not_real.pi
+        assert False  # Should not get here
+    except ModuleNotFoundError:
+        pass
+    assert isinstance(anything_not_real, DelayedImportErrorModule)
+    # see if it changes for second access
+    try:
+        anything_not_real.pi
+        assert False  # Should not get here
+    except ModuleNotFoundError:
+        pass
+
+
+def test_lazy_package_import():
```

Comment:
This is nice; would be good to get something like that over to the skimage PR too!

Reply:
> This is nice; would be good to get something like that over to the skimage PR too!

I'm going to try to get something like these tests of `attach` over in the skimage PR as suggested. 

Should I also update the `lazy.load` function there to allow "delayed failed modules"?  It currently raises a `ModuleNotFoundError` immediately upon `lazy.load` when a library is not found. That works well for many cases, but not if you want to import optional libraries. Thoughts @stefanv ?

Reply:
I think that would be a good feature to have, thanks Dan!